### DISCUSSION TITLE
安装总是失败

### DISCUSSION THREAD

**Discussion by choudou** [openelb/openelb#328] [Q&A]
安装总是失败

服务器为阿里云ECS
k8s是通过kubekey安装的单机版
使用https://raw.githubusercontent.com/openelb/openelb/master/deploy/openelb.yaml安装的,并修改了国内镜像

问题:
![image](https://github.com/openelb/openelb/assets/8089031/b47a9fd9-ec82-4337-bcf6-1897b0dda8b8)
安装完后 openelb-manager pod  总是 `ContainerCreating` 状态

原yaml文件中为`secretName: openelb-admission`,会报错找不到secret
![image](https://github.com/openelb/openelb/assets/8089031/60171d96-592e-470b-b40d-7c69660c7fa2)

改为具体的名字 `secretName: openelb-admission-token-mrq6d`
![image](https://github.com/openelb/openelb/assets/8089031/4eec960c-995a-4dd7-b279-0e2343201592)
后, 
![image](https://github.com/openelb/openelb/assets/8089031/a4c9554b-3fcd-4a53-ae98-a78b86a0a898)
报错变为和secret里的值不对应
![image](https://github.com/openelb/openelb/assets/8089031/26add7eb-ecfc-401e-980f-3d8f565bfaf7)
secret内容为
![image](https://github.com/openelb/openelb/assets/8089031/df631df6-899a-439c-a2fc-864d29c41fb0)

而且重复apply会提示 `0/1 nodes are available: 1 node(s) didn't have free ports for the requested pod ports.`

处理了2天了就是解决不了, 请求帮助,谢谢

**Comment by renyunkang** [comment]
https://github.com/openelb/openelb/blob/v0.5.1/deploy/openelb.yaml 先用这个试一下吧  master 会有一些正在开发的新特性，可能会出问题

**Comment by beyond03312** [comment]
你好，创建地址池的时候提示,这个怎么处理呢:
Error from server (InternalError): error when creating "eip-pool.yaml": Internal error occurred: failed calling webhook "validate.eip.network.kubesphere.io": Post "https://openelb-admission.openelb-system.svc:443/validate-network-kubesphere-io-v1alpha2-eip?timeout=10s": x509: certificate is valid for ingress.local, not openelb-admission.openelb-system.svc
/

**Reply by renyunkang** [reply]
> 你好，创建地址池的时候提示,这个怎么处理呢: Error from server (InternalError): error when creating "eip-pool.yaml": Internal error occurred: failed calling webhook "validate.eip.network.kubesphere.io": Post "https://openelb-admission.openelb-system.svc:443/validate-network-kubesphere-io-v1alpha2-eip?timeout=10s": x509: certificate is valid for ingress.local, not openelb-admission.openelb-system.svc /

证书错误，openelb-admission job 大概率是没有正常跑完，或者镜像拉不下来没启动起来。检查 openelb-admission job 的状态是否为 Completed

**Reply by beyond03312** [reply]
你好，是看这个状态吗
![image](https://github.com/openelb/openelb/assets/49865271/7626e31a-fef2-41f9-9262-55f4f08b9301)
![image](https://github.com/openelb/openelb/assets/49865271/35d8e8e8-9012-4503-9ab1-524bc0c12f84)
@renyunkang 麻烦您给看看，谢谢

**Reply by renyunkang** [reply]
> 你好，是看这个状态吗 ![image](https://user-images.githubusercontent.com/49865271/240777782-7626e31a-fef2-41f9-9262-55f4f08b9301.png) ![image](https://user-images.githubusercontent.com/49865271/240777818-35d8e8e8-9012-4503-9ab1-524bc0c12f84.png) @renyunkang 麻烦您给看看，谢谢

openelb-admission-patch job 失败了，可以自己排查一下是什么原因导致的

**Reply by beyond03312** [reply]
你好，我重新执行了一下安装脚本
![image](https://github.com/openelb/openelb/assets/49865271/a5f6fd66-d99c-4725-821d-8d07396860c1)
![image](https://github.com/openelb/openelb/assets/49865271/a5bc5eb3-f750-451c-9167-ac5374cebe52)
执行创建地址池还是报错，那个x509证书是怎么搞，跟openelb有什么关系，我看openelb的官方文档也没有提及
![image](https://github.com/openelb/openelb/assets/49865271/0b69ba67-dd00-4d2a-bfb8-7878eceba579)
